### DISCUSSION TITLE
Different results of `slice <string> <from> <to>` with too large <to>

### DISCUSSION THREAD

**Discussion by programandala-net** [arturo-lang/arturo#1282] [Q&A]
Why `slice "0123" 2 4` and `slice "0123" 2 5` give different results?:

```
$> slice "0123" 2 3
=> 23

$> slice "0123" 2 4
=> 23

$> slice "0123" 2 5
=>

```
In both cases the third parameter is beyond the last index. Therefore I expected the same result:  the final part or the string ("23") or an empty string (which is what `slice "0123" 2 5` returns), or `null`, or even an error.

Is it a bug or am I missing something?

**Comment by RickBarretto** [comment]
Yes, it's a bug! As you can see, the highest possible index in this case is 3.

Both, 4 and 5 are out-of bound indexes, so this should be illegal. 
We have two options to fix that: 1) Raise an exception 2) Clip-it and just consider the possible index (lowest and highest).
